programs/cleos/main.cpp

```diff
@@ -512,14 +514,20 @@ void send_transaction( signed_transaction& trx, int32_t extra_kcpu, packed_trans
    }
 }
 
-chain::action create_newaccount(const name& creator, const name& newaccount, public_key_type owner, public_key_type active) {
+chain::permission_level get_permission_level(const std::string& str) {
+   auto at_pos = str.find('@');
+   EOSC_ASSERT(at_pos != std::string::npos, "invalid permission level");
+   return permission_level { str.substr(0, at_pos), str.substr(at_pos + 1) };
+}
+
+chain::action create_newaccount(const name& creator, const name& newaccount, auth_type owner, auth_type active) {
    return action {
       get_account_permissions(tx_permission, {creator,config::active_name}),
       eosio::chain::newaccount{
          .creator      = creator,
          .name         = newaccount,
-         .owner        = eosio::chain::authority{1, {{owner, 1}}, {}},
-         .active       = eosio::chain::authority{1, {{active, 1}}, {}}
+         .owner        = !owner.which() ? authority(owner.get<public_key_type>()) : authority(owner.get<permission_level>()),
```

Comment:
please use `owner.contains<public_key_type>()` so that its explicit and durable to reordering instead of implicitly depending on the falsey nature of the `0` value returned from `which()`